Feat admin dashboard

laurarvegav opened the pull request:
Together we added features for Admin Dashboard to display the correct header, links to admin merchants and invoices indexes, and the top 5 customers with the number of successful transactions conducted by them.

For this we settled up some factory files for testing purposes and also two methods in Customer model with the corresponding tests.

All tests passing.

Note: we just noticed we need to add the setup for SimpleCov.

closes #27 
closes #24 
closes #21

JackCSweeney commented on app/views/admin/index.html.erb:
Had a thought on this last night, and maybe you're already thinking about it this way, but I think as we go through building out these different views, we can make sure we get everything that we need present on the view from the wire frame, then worry about styling it to align with how the wireframe looks once everything is already there. Might make it a bit smoother and so we don't get stuck or waste time in a styling rabbit hole while we still have more things that we'll be adding to the view later.

JackCSweeney commented on app/models/customer.rb:
```diff
@@ -3,4 +3,13 @@ class Customer < ApplicationRecord
   has_many :invoices, dependent: :destroy
+  has_many :transactions, through: :invoices
+
+  def self.top_customers
```
Should we potentially change this method name to 'top_five_customers'? Or does that feel unnecessary?

laurarvegav replied:
This is done, I changed it everywhere

JackCSweeney commented on spec/factories/invoices.rb:
```diff
@@ -1,6 +1,11 @@
   factory :invoice do
-    status { 1 }
-    customer { nil }
+    id { Faker::Number.unique.number }
```
Looks like we missed this spot as a place that we can remove the id attribute from our factory bot since we figured out the create method will let it generate the id, created at, etc.

laurarvegav replied:
We found another one and took it out, not needed after using the create_list instead of build_list.

JackCSweeney reviewed (commented):
Added a few comments of potential changes we could do before or after the merge, just wanted to double check if we wanted to do them before hand.

JackCSweeney commented:
Amazing! Thank you, Laura. We're cruising now!

laurarvegav commented:
We just went over the suggestions and also added SimpleCov setup and added the variables to identify customers 1 to 4 inside of admin dashboard spec file.
